noir-projects/aztec-nr/aztec/src/context/private_context.nr

```diff
@@ -489,28 +461,37 @@ impl PrivateContext {
         is_static_call: bool,
         is_delegate_call: bool
     ) {
+        let counter = self.next_counter();
```

Comment:
I notice that we now increment the counter _before_ setting it as the `.counter` of a public call (inc teardown), does this mean we 'skip' a counter or was it incorrect before?

Reply:
The naming is confusing. `next_counter` actually returns the current counter 😂

Reply:
Oh yep so it does!